plugins/Kaleidoscope-Colormap-Overlay/src/kaleidoscope/plugin/Colormap-Overlay.cpp

```diff
@@ -30,20 +30,39 @@
 #include "kaleidoscope/key_defs.h"              // for Key, KEY_FLAGS, Key_NoKey, LockLayer
 #include "kaleidoscope/layers.h"                // for Layer, Layer_
 #include "kaleidoscope/plugin/LEDControl.h"     // for LEDControl
+#include <Kaleidoscope.h>                       // for Kaleidoscope
+#include <Kaleidoscope-FocusSerial.h>           // for Focus
 #include <Kaleidoscope-LED-Palette-Theme.h>     // for LEDPaletteTheme
+#include <Kaleidoscope-EEPROM-Settings.h>       // for EEPROMSettings
 
 namespace kaleidoscope {
 namespace plugin {
+uint16_t ColormapOverlay::overlays_base_;
+Overlay *ColormapOverlay::overlays_;
+uint8_t ColormapOverlay::overlay_count_;
+
 void ColormapOverlay::setup() {
   ::LEDPaletteTheme.reservePalette();
+
+  overlays_base_ = ::EEPROMSettings.requestSlice(max_overlays_ * sizeof(Overlay));
+
+  ::EEPROMSettings.seal();
+
+  if (!::EEPROMSettings.isValid()) {
+    // When settings are invalid, better to clean it all up to prevent unwanted things from happening
+    ::EEPROMSettings.invalidate();
```

Comment:
Re: https://github.com/keyboardio/Kaleidoscope/pull/1434#discussion_r2038431488

Is this the correct way to handle this?

Reply:
I don't believe so.  the seal is taken care of automatically. and isValid is a whole-system level thing.
This should be the right pattern
```
EventHandlerResult Keyclick::onSetup() {
  // Request storage space and load settings
  bool success = ::EEPROMSettings.requestSliceAndLoadData(&settings_base_, &settings_);

  // If settings weren't loaded successfully, use defaults
  if (!success) {
    settings_.enabled = true;
    settings_.version = 0x01;
    updateSettings();
  }
```